Dockerfile

```
@@ -0,0 +1,57 @@
+# Switch to non-root user
+USER crawler
+
+# Expose ports
+EXPOSE 8010 38080
+
+# Health check for both services
+HEALTHCHECK --interval=30s --timeout=10s --start-period=30s --retries=3 \
+    CMD curl -f http://localhost:8010/health && curl -f http://localhost:38080/health || exit 1
+
+# Run supervisor to manage both services
+USER root
```

Hadolint (2.12.0)

[warning] Last USER should not be root

(DL3002)